src/EventLoop.php

```diff
@@ -391,10 +391,11 @@ public static function run(): void
             self::$fiber = self::createFiber();
         }
 
-        if (self::$fiber->isStarted()) {
-            self::$fiber->resume();
-        } else {
-            self::$fiber->start();
+        $lambda = self::$fiber->isStarted() ? self::$fiber->resume() : self::$fiber->start();
+
+        if ($lambda) {
+            $lambda();
+            throw new \RuntimeException('Interrupt from event loop must throw an exception');
```

Comment:
Shouldn't we throw an `\Error` here?

Reply:
My thinking was it shouldn't ever happen unless _we_ did something wrong, but being then it originated from a programming error I agree, `\Error` does seem more appropriate.